### PR TITLE
Contribute lsp4mp.ls jars

### LINKED ISSUE #290: Contribute lsp4mp.ls jars
quarkus.ls.ext jar should be referenced in package.json so it can be picked up by vscode-microprofile

### PR DESCRIPTION
**Pull request by angelozerr**
Contribute lsp4mp.ls jars

Fixes #290

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
To test this PR:

 * uninstall vscode-quarkus
 * clone https://github.com/redhat-developer/vscode-microprofile
 * create a vsix for vscode-microprofile
 * install the vscode-microprofile vsix 
 * import this PR in vscode-quarkus and check everything is working (generate Quarkus project, completion for application.properties and Java, snippets for Quarkus in Java and properties)

File: package.json [hunk 1]
```diff
@@ -47,25 +47,20 @@
...
     "vscjava.vscode-java-debug"
   ],
   "contributes": {
     "javaExtensions": [
-      "./jars/org.eclipse.lsp4mp.jdt.core.jar",
       "./jars/com.redhat.microprofile.jdt.quarkus.jar"
     ],
+    "microprofile": {
+      "jarExtensions": [
+        "./server/com.redhat.quarkus.ls-0.9.0-SNAPSHOT.jar"
```
Comment (angelozerr) [inline package.json]: @fbricon if you can give me quickly how to generate this JAR without 0.9.0-SNAPSHOT it would be nice.

**Comment by fbricon** [comment]
Travis build fails:
```
ERROR in /home/travis/build/redhat-developer/vscode-quarkus/src/QuarkusProjectListener.ts
./src/QuarkusProjectListener.ts
[tsl] ERROR in /home/travis/build/redhat-developer/vscode-quarkus/src/QuarkusProjectListener.ts(18,28)
      TS2307: Cannot find module 'vscode-languageclient'.
ERROR in /home/travis/build/redhat-developer/vscode-quarkus/src/extension.ts
./src/extension.ts
[tsl] ERROR in /home/travis/build/redhat-developer/vscode-quarkus/src/extension.ts(18,91)
      TS2307: Cannot find module 'vscode-languageclient'.
ERROR in /home/travis/build/redhat-developer/vscode-quarkus/src/wizards/debugging/terminateProcess.ts
./src/wizards/debugging/terminateProcess.ts
[tsl] ERROR in /home/travis/build/redhat-developer/vscode-quarkus/src/wizards/debugging/terminateProcess.ts(18,28)
      TS2307: Cannot find module 'vscode-languageclient'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! vscode-quarkus@1.7.0 vscode:prepublish: `webpack --mode production`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the vscode-quarkus@1.7.0 vscode:prepublish script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2020-07-16T15_51_56_180Z-debug.log
The command "npm run vscode:prepublish" failed and exited with 2 during .
```

File: src/extension.ts [hunk 2]
```diff
@@ -13,28 +13,21 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import * as requirements from './languageServer/requirements';
-
 import * as path from 'path';
-import { VSCodeCommands, MicroProfileLS } from './definitions/constants';
+import { VSCodeCommands } from './definitions/constants';
 import { DidChangeConfigurationNotification, LanguageClientOptions, LanguageClient } from 'vscode-languageclient';
```
Comment (xorye) [inline src/extension.ts]: Line 18 should be removed

File: src/QuarkusProjectListener.ts [hunk 3]
```diff
@@ -17,7 +17,16 @@ import { commands, workspace } from 'vscode';
 import { getWorkspaceProjectLabels } from './utils/workspaceUtils';
 import { Disposable } from 'vscode-languageclient';
```
Comment (xorye) [inline src/QuarkusProjectListener.ts]: Remove this line, and import `Disposable` from `vscode` on line 16:

```
import { commands, workspace, Disposable } from 'vscode';
```

Comment (xorye) [reply src/QuarkusProjectListener.ts]: Same for https://github.com/angelozerr/vscode-quarkus/blob/7c41108b79144231902c10b5fb502009cb74edc4/src/wizards/debugging/terminateProcess.ts#L18

**Comment by xorye** [comment]
This dependency can be removed: https://github.com/redhat-developer/vscode-quarkus/blob/7c41108b79144231902c10b5fb502009cb74edc4/package.json#L279

File: src/extension.ts (hunk 2 above)
Comment (angelozerr) [reply src/extension.ts]: fixed

File: src/QuarkusProjectListener.ts (hunk 3 above)
Comment (angelozerr) [reply src/QuarkusProjectListener.ts]: fixed

**Comment by angelozerr** [comment]
> This dependency can be removed:

fixed

File: gulpfile.js [hunk 4]
```diff
@@ -46,15 +32,7 @@ gulp.task('buildQuarkusServerExt', (done) => {
...
-gulp.task('buildServer', gulp.series(['buildMicroProfileServer', 'buildQuarkusServerExt']));
-
-gulp.task('buildMicroProfileExtension', (done) => {
-  cp.execSync(mvnw() + ' -pl "' + microprofileExtension + '" clean verify -DskipTests', { cwd: microprofileExtensionDir, stdio: 'inherit' });
-  gulp.src(microprofileExtensionDir + '/' + microprofileExtension + '/target/' + microprofileExtension + '-!(*sources).jar')
-    .pipe(rename(microprofileExtension + '.jar'))
-    .pipe(gulp.dest('./jars'));
-  done();
-});
+gulp.task('buildServer', gulp.series(['buildQuarkusServerExt']));
```
Comment (xorye) [inline gulpfile.js]: I think its a bit redundant to have this. We can delete this task and rename https://github.com/redhat-developer/vscode-quarkus/blob/7b1c4a2fbdba0c6216e1d16bde7b8d0786360fc3/gulpfile.js#L24 to 
```
gulp.task('buildServer', (done) => {
```

File: gulpfile.js [hunk 5]
```diff
@@ -64,7 +42,7 @@ gulp.task('buildQuarkusExtension', (done) => {
   done();
 });
 
-gulp.task('buildExtension', gulp.series(['buildMicroProfileExtension', 'buildQuarkusExtension']));
+gulp.task('buildExtension', gulp.series(['buildQuarkusExtension']));
```
Comment (xorye) [inline gulpfile.js]: Same for this task

**Comment by xorye** [comment]
> import this PR in vscode-quarkus and check everything is working (generate Quarkus project, completion for application.properties and Java, snippets for Quarkus in Java and properties)

I couldn't find any regression

**Comment by angelozerr** [comment]
> Travis build fails:

fixed (thanks @xorye ) but there are a problem with timeout test. @xorye have you some idea?

File: gulpfile.js [hunk 6]
```diff
@@ -15,27 +15,13 @@ const gulp = require('gulp');
...
-  cp.execSync(mvnw() + ' clean install -DskipTests', { cwd: microprofileServerDir , stdio: 'inherit' });
-  gulp.src(microprofileServerDir + '/target/' + microprofileServerName)
-    .pipe(gulp.dest('./server'));
-  done();
-});
-
-gulp.task('buildQuarkusServerExt', (done) => {
+gulp.task('buildServer', (done) => {
   cp.execSync(mvnw() + ' clean verify -DskipTests', { cwd: quarkusServerExtDir , stdio: 'inherit' });
   gulp.src(quarkusServerExtDir + '/target/' + quarkusServerExtGlob)
```
Comment (fbricon) [inline gulpfile.js]: .pipe(rename('com.redhat.quarkus.ls.jar'))

File: package.json (hunk 1 above)
Comment (angelozerr) [reply package.json]: fixed

**Comment by angelozerr** [comment]
> This dependency can be removed:

fixed

File: gulpfile.js (hunk 6 above)
Comment (angelozerr) [reply gulpfile.js]: I follow this idea but using const, fixed.

File: gulpfile.js (hunk 4 above)
Comment (angelozerr) [reply gulpfile.js]: fixed

File: gulpfile.js (hunk 5 above)
Comment (angelozerr) [reply gulpfile.js]: fixed

**Comment by xorye** [comment]
One way to disable UI tests quickly is to change `npm run test-all` to `npm test` here:
https://github.com/redhat-developer/vscode-quarkus/blob/58eada4877a47dcf7249869b733581988f74dbb5/.travis.yml#L36.

The jenkins file already does this:
https://github.com/redhat-developer/vscode-quarkus/blob/58eada4877a47dcf7249869b733581988f74dbb5/Jenkinsfile#L50

**Comment by angelozerr** [comment]
> One way to disable UI tests quickly is to change npm run test-all to npm test here:

Thanks fixed.

I have updated CONTRIBUTING just to add reference to vscode-microprofile but it should be improved again. I think the whole content which speaks about MP LS must be removed and hosted in vscode-microprofile. I have an issue for that https://github.com/redhat-developer/vscode-microprofile/issues/2

File: README.md [hunk 7]
```diff
@@ -70,6 +68,8 @@ In `Java` files, you will benefit with:
 
 The following settings are supported:
 
+* See all [microprofile.tools.* settings](https://github.com/redhat-developer/vscode-microprofile#supported-vs-code-settings) for `MicroProfile/Quarkus` support in `properties` and `Java` files.
```
Comment (xorye) [inline README.md]: Can we change this too:

* All [microprofile.tools.* settings](https://github.com/redhat-developer/vscode-microprofile#supported-vs-code-settings) from the [Visual Studio Code extension for MicroProfile](https://github.com/redhat-developer/vscode-microprofile)

**Comment by xorye** [comment]
I think we should add vscode microprofile in the readme requirements:
https://github.com/redhat-developer/vscode-quarkus/blame/ffae27b032a5c750ebded4fc0bcac2bddcb22833/README.md#L61

File: README.md (hunk 7 above)
Comment (angelozerr) [reply README.md]: fixed

**Comment by angelozerr** [comment]
>I think we should add vscode microprofile in the readme requirements:
https://github.com/redhat-developer/vscode-quarkus/blame/ffae27b032a5c750ebded4fc0bcac2bddcb22833/README.md#L61

fixed

**Comment by xorye** [comment]
I haven't had problems with the yaml completion, but yaml value validation doesn't seem to be working.

If I have a boolean config property:
```
    @ConfigProperty(name = "greeting.myBool") 
    boolean bool;
```

I do not have validation:
image.

I am using the latest version of vscode-yaml and I have `yaml.validate = true` in the settings

**Comment by xorye** [comment]
Oops, nevermind. I don't have a problem with yaml value validation

**Pull request merged**